back/src/adapters/secondary/pg/repositories/PgAgencyRepository.integration.test.ts

```diff
@@ -656,6 +673,32 @@ describe("PgAgencyRepository", () => {
       expect(inDb).toHaveLength(1);
       expectToEqual(inDb[0], { ...agency1, status: "active" });
     });
+
+    it("throws an error if agency already exists with same information", async () => {
+      await agencyRepository.insert(agency1);
+      await agencyRepository.insert(agency2);
+      agency2.kind = agency1.kind;
+      agency2.address = {
+        ...agency1.address,
+      };
```

Comment:
On essaye d'éviter les mutations. Par exemple, on peut utiliser les builders :

```suggestion
      await agencyRepository.insert(agency1);
      await agencyRepository.insert(agency2); // pas sur que ça soit necessaire de l'insérer d'ailleur non ?
      const agency2again = agency1builder.withId(agency2.id).withKind(agency1.kind).withAddress(agency1.address)
```